Lesson 1/script.js

```diff
@@ -0,0 +1,166 @@
+        super(name,age,dateOfBirth);
+        this.salary = salary;
+        this.department = department;
+    }
+    _displayInfo(){
+        // return `${super._displayInfo()} ${this.salary} ${this.department}`
+        return{'Human_class':super._displayInfo(), 'salary':this.salary,'department':this.department};
+    }
+}
+class Developer extends Employee{
+    constructor(name,age,dateOfBirth, salary = 0, department,manager){
+        super(name,age,dateOfBirth,salary,department);
+        this.manager = manager;
+    }
+    _displayInfo(){
+        return {'Employee_class':super._displayInfo(), 'manager':this.manager};
+    }
+    _changeManager(newManager){
+        this.manager = newManager;
+    }
+}
+class Manager extends Employee{
+    constructor(name,age,dateOfBirth, salary = 0, department, ...employees){
+        super(name,age,dateOfBirth,salary, department);
+    this.employees = employees;
+    }
+    _addDeveloper(employee){
+        this.employees.push(employee);
+    }
+    _removeDeveloper(...employee){
+        console.log(employee);
+        employee.forEach((item)=>{
+            this.employees = this.employees.filter((e)=>e !== item);
+        })
+    }
+    _displayInfo(){
+        return {'Employee_class':super._displayInfo(), 'employees':this.employees};
+    }
+}
+//новый человек
+let JohnDoe = new Human('John Doe',19,'2000-01-01');
+console.log(JohnDoe);
+console.log('-------');
+//новый сотрудник
+let MartaClaus = new Employee('Marta Claus',34,'1985-01-01', 1000, 'Financial');
+console.log(MartaClaus);
+console.log('-------');
+//новый разработчик
+let AlanFitzgerald= new Developer('Alan Fitzgerald',25,'1995-01-01',2000,'IT','Anne Frank');
+console.log(AlanFitzgerald);
+console.log('-------');
+//новый руководитель
+let AnneFrank= new Manager('Anne Frank', 29, '1990-01-01',1500,'Management','Alan Fitzgerald','Rocky Balboa','Harry Houdini');
+console.log(AnneFrank);
+console.log('-------');
+//добавляем сотрудника
+AnneFrank._addDeveloper('New Dev');
+console.log(AnneFrank._displayInfo());
+console.log('-------');
+//удаляем сотрудников
+AnneFrank._removeDeveloper('Harry Houdini','Rocky Balboa');
+console.log(AnneFrank._displayInfo());
+console.log('-------');
+//меняем руководителя
+AlanFitzgerald._changeManager('Holly Dolly');
+console.log(AlanFitzgerald._displayInfo());
+console.log('-------');
+
+//Task5
+console.log('Task 5');
+//Реализация с промисами напрямую
+let link='https://jsonplaceholder.typicode.com/users/';
+for (let i = 1; i<=10; i++){
+    let fetchingPost = new Promise((res,rej)=>{
+        $.get(`${link}${i}`)
+            .done(posts=>res(posts))
+            .fail(err=>rej(err));
+    });
+    fetchingPost
+        .then(posts=>console.log(posts))
+        .catch(err=>console.log(err));
+}
+console.log('------------------');
+//Реализация с фетчем
+let resultList = [];
+for (let i = 1; i<=10; i++){
+    fetch((`${link}${i}`))
+        .then(result=>resultList.push(result))
```

Comment:
```suggestion
        .then(result=>result.json()).then(json => resultList.push(json))
```
так мы передадим сам json-объект в resultList, а не заголовки (result) запроса:)